resources/airflow.rb

```diff
@@ -1708,36 +1710,34 @@ def self.create_infil_mech_vent_objects(model, building, infil, mech_vent, wind_
       infil_program.addLine("    Set QWHV = #{mech_vent.cfis_f_damper_open_var.name}*CFIS_Q_duct")
-      infil_program.addLine("    Set cfistemp2 = #{mech_vent.cfis_f_damper_open_var.name}*(ZoneTimeStep*60)")
-      infil_program.addLine("    Set #{mech_vent.cfis_t_sum_open_var.name} = #{mech_vent.cfis_t_sum_open_var.name}+cfistemp2")
+      infil_program.addLine("    Set #{mech_vent.cfis_t_sum_open_var.name} = #{mech_vent.cfis_t_sum_open_var.name}+cfis_fan_runtime")
       infil_program.addLine("    Set cfis_cfm = airloop_cfm*#{mech_vent.cfis_airflow_frac}")
-      infil_program.addLine("    Set cfis_frac = #{mech_vent.cfis_f_damper_open_var.name}*(1-fan_rtf)")
-      infil_program.addLine("    Set #{whole_house_fan_actuator.name} = CFIS_fan_power*cfis_cfm*cfis_frac")
+      infil_program.addLine("    Set #{mech_vent.cfis_f_damper_extra_open_var.name} = @Max (#{mech_vent.cfis_f_damper_open_var.name}-fan_rtf_hvac) 0.0") # additional runtime fraction to meet min/hr
+      infil_program.addLine("    Set #{whole_house_fan_actuator.name} = CFIS_fan_power*cfis_cfm*#{mech_vent.cfis_f_damper_extra_open_var.name}")
       infil_program.addLine("  Else")
-      infil_program.addLine("    Set cfistemp4 = fan_rtf*ZoneTimeStep*60")
-      infil_program.addLine("    If (#{mech_vent.cfis_t_sum_open_var.name}+cfistemp4) > CFIS_t_min_hr_open")
+      # No need to turn on blower for extra ventilation
+      infil_program.addLine("    Set cfis_fan_runtime2 = fan_rtf_hvac*ZoneTimeStep*60")
+      infil_program.addLine("    If (#{mech_vent.cfis_t_sum_open_var.name}+cfis_fan_runtime2) > CFIS_t_min_hr_open")
       # Damper is only open for a portion of this time step to achieve target minutes per hour
-      infil_program.addLine("      Set cfistemp5 = CFIS_t_min_hr_open-#{mech_vent.cfis_t_sum_open_var.name}")
-      infil_program.addLine("      Set #{mech_vent.cfis_f_damper_open_var.name} = cfistemp5/(ZoneTimeStep*60)")
+      infil_program.addLine("      Set cfis_fan_runtime = CFIS_t_min_hr_open-#{mech_vent.cfis_t_sum_open_var.name}")
+      infil_program.addLine("      Set #{mech_vent.cfis_f_damper_open_var.name} = cfis_fan_runtime/(ZoneTimeStep*60)")
       infil_program.addLine("      Set QWHV = #{mech_vent.cfis_f_damper_open_var.name}*CFIS_Q_duct")
       infil_program.addLine("      Set #{mech_vent.cfis_t_sum_open_var.name} = CFIS_t_min_hr_open")
       infil_program.addLine("    Else")
       # Damper is open and using call for heat/cool to supply fresh air
-      infil_program.addLine("      Set cfistemp6 = fan_rtf*ZoneTimeStep*60")
-      infil_program.addLine("      Set #{mech_vent.cfis_t_sum_open_var.name} = #{mech_vent.cfis_t_sum_open_var.name}+cfistemp6")
-      infil_program.addLine("      Set #{mech_vent.cfis_f_damper_open_var.name} = 1")
-      infil_program.addLine("      Set QWHV = fan_rtf*CFIS_Q_duct")
+      infil_program.addLine("      Set cfis_fan_runtime = fan_rtf_hvac*ZoneTimeStep*60")
```

Comment:
I've changed some names of EMS variables to help myself better understand the program. If they're not appropriate or making EMS harder to read, please let me know.